### PR TITLE
Issue #929: Changed private static final loggers to private final ones

### PR DESCRIPTION
**Pull request by samuel-gu**
This pull request is for changing private static final loggers in codebase to private final following Issue #929

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 1]
```diff
@@ -66,7 +66,7 @@
     public static final String EXCEPTION_MSG = "general.exception";
 
     /** Logger for Checker. */
-    private static final Log LOG = LogFactory.getLog(Checker.class);
+    private final Log log;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: restore definition of logger in field.
The same for all other cases

Comment (samuel-gu) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: I've restored definition of logger in field for all cases. Should I still update the main?

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: yes

**Comment by rnveach** [comment]
@samuel-gu ping
Since we are out of proposal deadline, do you think you can finish this?

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java (hunk 1 above)
Comment (samuel-gu) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: Updating the main class' logger variable has these difficulties.
1. All the methods in the main that assess the log variable are defined as static
2. The main class cannot be instantiated to call the non-static log in the methods that call the log.

Any thought on what to do?

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: @samuel-gu ,
keep it static, Main is not part of checkstyle as library so, static usage is completely ok in it.

Comment (samuel-gu) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: Then there are no other classes with private static final definition for loggers. Looks like there wasn't an issue after all

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: ok, please focus on my point "2)"  to enforce this by Checkstyle rule , Main.java should be added to suppression https://github.com/checkstyle/checkstyle/blob/master/config/sevntu_suppressions.xml

**Comment by romani** [comment]
base on discussion around the code ..... my point "1." should be ignored.

**Comment by romani** [comment]
@samuel-gu , let finish this PR, looks like we are in one step from success.

**Comment by rnveach** [comment]
@samuel-gu why did you close the PR and delete the branch?

**Comment by samuel-gu** [comment]
@rnveach I was having issues with my local repository, so i re-cloned.

**Comment by rnveach** [comment]
@samuel-gu will you re-open or recreate this PR then?

**Pull request closed without merging**